core-v1/src/main/scala/algebra/ExprConverter.scala

```diff
@@ -0,0 +1,50 @@
+package com.rallyhealth.vapors.v1
+
+package algebra
+
+import data.Window
+
+import shapeless.ops.hlist.Tupler
+import shapeless.{Generic, HList}
+
+/**
+  * A serializable description of a total conversion function from one type to another.
+  */
+sealed trait ExprConverter[-I, +O] extends (I => O) {
+
+  /**
+    * The type of conversion applied.
+    */
+  def conversionType: String
+
+  /**
+    * Apply the conversion.
+    *
+    * @note this should never throw an exception.
+    */
+  override def apply(inputValue: I): O
+}
+
+object ExprConverter {
+
+  // TODO: Capture type information about R for debugging?
+  private final class Impl[-I, +O](
+    override val conversionType: String,
+    convert: I => O,
+  ) extends ExprConverter[I, O] {
+    override def apply(in: I): O = convert(in)
+    override val toString: String = s"ExprConverter.$conversionType"
+  }
```

Comment:
The type name is now `O` and I am not sure I will actually use this converter because I have to handle wrapped types which becomes a lot more complicated. This pattern only handles types of kind `*` and not `* -> *` or constants.